lib/src/native/realm_core.dart

```diff
@@ -2106,6 +2106,36 @@ class _RealmCore {
       return completer.future;
     });
   }
+
+  static void _call_app_function_callback(Pointer<Void> userdata, Pointer<Char> response, Pointer<realm_app_error> error) {
+    final Completer<String>? completer = userdata.toObject(isPersistent: true);
+    if (completer == null) {
+      return;
+    }
+    if (error != nullptr) {
+      completer.completeWithAppError(error);
+      return;
+    }
+
+    final stringResponse = response.cast<Utf8>().toRealmDartString()!;
+    completer.complete(stringResponse);
+  }
+
+  Future<String> callAppFunction(App app, User user, String functionName, String? argsAsJSON) {
+    return using((arena) {
+      final completer = Completer<String>();
```

Comment:
Should this be a `CancellableCompleter` instead?

Reply:
We are not going to provide cancellation option here.